packages/blob-storage-manager/src/BlobStorageManager.ts

```diff
@@ -90,29 +113,27 @@ export class BlobStorageManager<
     const failedUploads = results.filter(
       (res) => res.status === "rejected"
     ) as PromiseRejectedResult[];
-    const storageErrors = failedUploads.map((res, i) => {
+    const storageErrors = failedUploads.map<StorageError<SNames>>((res, i) => {
       const storage = availableStorages[i] as [SNames, BlobStorage];
       const storageName = storage[0];
 
-      return `-${storageName}: ${res.reason}`;
+      return {
+        storage: storageName,
+        error: res.reason,
+      };
     });
 
     if (!successfulUploads.length) {
       throw new Error(
-        `Failed to upload blob to any of the storages :\n${storageErrors.join(
-          "\n"
-        )}`
-      );
-    }
-
-    if (failedUploads.length) {
-      console.warn(
-        `Couldn't upload blob to some of the storages :\n ${storageErrors.join(
+        `Failed to upload blob ${versionedHash} to any of the storages :\n${storageErrors.join(
           "\n"
```

Comment:
In my experience when debugging issues and reviewing logs, this \n will make it less readable. Do you mind joining with commas? (", ")

Reply:
Sure, no problem.